File: blog-backend/src/main/java/com/aicogniblog/guestbook/entity/Guestbook.java
package com.aicogniblog.guestbook.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 留言板实体类
 */
@Data
public class Guestbook {
    private Long id;
    private Long userId;
    private String name;
    private String email;
    private String content;
    private String avatarUrl;
    private String ipAddress;
    private Integer status;  // 0=待审核 1=已发布 2=已拒绝
    private Integer deleted;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: blog-frontend/src/api/guestbook.ts
import http from './http';

export interface GuestbookMessage {
  id: number;
  userId?: number;
  name: string;
  email?: string;
  content: string;
  avatarUrl?: string;
  createdAt: string;
}

export interface GuestbookRequest {
  name: string;
  email?: string;
  content: string;
}

export const guestbookApi = {
  // 获取留言列表
  getMessages: (params: { page?: number; size?: number }) =>
    http.get<{ content: GuestbookMessage[]; totalElements: number }>('/guestbook', { params }),
  
  // 发表留言
  postMessage: (data: GuestbookRequest) =>
    http.post('/guestbook', data),
  
  // 删除留言（管理员）
  deleteMessage: (id: number) =>
    http.delete(`/guestbook/${id}`),
};

